File: EstruturaDeDados&Algoritmos/B-Arrays/OutrasLinguagens/Compiladas/Array.zig
const std = @import("std");

pub fn main() !void {
    // Declaração: [Tamanho]Tipo
    // O '_' diz ao compilador: "conte os elementos para mim"
    const numeros = [_]i32{1, 2, 3};

    // Acessando um índice específico ({d} é para decimais/inteiros)
    std.debug.print("Primeiro valor: {d}\n", .{numeros[0]});

    // Imprimindo o array inteiro ({any} imprime qualquer estrutura)
    std.debug.print("Array completo: {any}\n", .{numeros});
}

File: EstruturaDeDados&Algoritmos/B-Arrays/OutrasLinguagens/Compiladas/Array.rs
fn main() {
    // Array de tamanho fixo [Tipo; Tamanho]
    let numeros: [i32; 3] = [10, 20, 30];
    
    // Para imprimir array inteiro, usa-se {:?} debug
    println!("{:?}", numeros); 

    //ou

     // Declaração implícita de tipo e tamanho
    let numerosInp = [1, 2, 3];

    // {:?} é usado para debug printing de arrays
    println!("{:?}", numerosInp);
}